mcda/utils/utils_for_main.py

```diff
@@ -0,0 +1,1200 @@
+                       iterative_random_w_score_means=iterative_random_w_score_means,
+                       iterative_random_w_score_means_normalized=iterative_random_w_score_means_normalized,
+                       iterative_random_w_score_stds=iterative_random_w_score_stds,
+                       index_column_name=index_column_name, index_column_values=index_column_values,
+                       input_config=input_config)
+
+    _plot_and_save_charts(scores=scores, normalized_scores=normalized_scores,
+                          score_means=all_weights_score_means, score_stds=all_weights_score_stds,
+                          score_means_normalized=all_weights_score_means_normalized,
+                          iterative_random_w_score_means=iterative_random_w_score_means,
+                          iterative_random_w_score_stds=iterative_random_w_score_stds,
+                          iterative_random_w_score_means_normalized=iterative_random_w_score_means_normalized,
+                          input_matrix=input_matrix, config=input_config,
+                          is_robustness_weights=is_robustness_weights)
+
+    elapsed = time.time() - t
+    logger.info("All calculations finished in seconds {}".format(elapsed))
+
+
+def run_mcda_with_indicator_uncertainty(input_config: dict, input_matrix: pd.DataFrame,
+                                        index_column_name: str, index_column_values: list,
+                                        mc_runs: int, is_sensitivity: str, f_agg: str, f_norm: str,
+                                        weights: Union[List[list], List[pd.DataFrame], dict],
+                                        polar: List[str], marginal_pdf: List[str]) -> None:
+    """
+    Runs ProMCDA with uncertainty on the indicators, i.e. with a robustness analysis.
+
+    This function executes the MCDA process with uncertainty in the indicators.
+    It computes scores, saves the results, and logs the completion time.
+
+    Parameters:
+    - input_matrix: The input_matrix without the alternatives.
+    - index_column_name: the name of the index column of the original input matrix.
+    - index_column_values: the values of the index column of the original input matrix.
+    - weights: the normalised weights (either fixed or random sampled weights, depending on the settings).
+      In the context of the robustness analysis, only fixed normalised weights are used, i.e. weights[0].
+
+    :return: None
+    :param input_config: dict
+    :param index_column_name: str
+    :param index_column_values: list
+    :param input_matrix: pd:DataFrame
+    :param mc_runs: int
+    :param is_sensitivity: str
+    :param weights: Union[List[str], List[pd.DataFrame], dict, None]
+    :param f_norm: str
+    :param f_agg: str
+    :param polar: List[str]
+    :param marginal_pdf: List[str]
+    """
+    logger.info("Start ProMCDA with uncertainty on the indicators")
```

Comment:
I did not run the code yet, but probably to have some `logger.info("********")` before and after the logger line can result in some nice output in the console?